core/editor/src/tools/select.rs

```diff
@@ -0,0 +1,42 @@
+use crate::events::Event;
+use crate::events::MouseKeys;
+use crate::tools::Tool;
+
+pub struct Select(Fsm);
+
+impl Tool for Select {
+	fn handle_input(&mut self, event: Event) {
+		match event {
+			Event::MouseDown(state) => {
+				if state.mouse_keys == MouseKeys::LEFT {
```

Comment:
I'm not to familiar with the bit vector crate but that seems like it would compare all bits but that might not be what we want

Reply:
Not sure how to fix this one.

Reply:
`if state.mouse_keys && MousKeys::LEFT == 1`